### PR TITLE
[Crypto] improve BLS compression settings in the lib

### PR DESCRIPTION
**Pull request by tarakby**
- generalize the tests to cover non-compressed signatures.
- separate the compression settings for signatures and public keys for a better flexibility of the lib.

### CODE REVIEW

File: crypto/thresholdsign_test.go [hunk 1]
```diff
@@ -15,10 +15,10 @@ import (
 )
 
 func TestThresholdSignature(t *testing.T) {
-	t.Run("stateful_simple_keygen", testStatefulThresholdSignatureSimpleKeyGen)
-	t.Run("stateful_signature_feldmanVSS_keygen", testStatefulThresholdSignatureFeldmanVSS)
-	t.Run("stateful_signature_jointFeldman_keygen", testStatefulThresholdSignatureJointFeldman)
-	t.Run("stateless_simple_keygen", testStatelessThresholdSignatureSimpleKeyGen)
+	//t.Run("stateful_simple_keygen", testStatefulThresholdSignatureSimpleKeyGen)
```
Comment (Kay-Zee) [inline crypto/thresholdsign_test.go:18]: Any reason we're commenting out all the tests here? If this is on purpose, should we add some reasoning to the comments?

File: crypto/types.go [hunk 2]
```diff
@@ -92,5 +98,7 @@ const (
 	maxRelicPrgSeed = 1 << 32
 )
 
+const ()
```
Comment (Kay-Zee) [inline crypto/types.go:101]: What's this for?

File: crypto/thresholdsign_test.go (hunk 1 above)
Comment (tarakby) [reply crypto/thresholdsign_test.go:18]: They should be uncommented, I didn't clean after debugging.

File: crypto/types.go (hunk 2 above)
Comment (tarakby) [reply crypto/types.go:101]: This was not cleaned either, thanks for pointing it out.

**Review by AlexHentschel — APPROVED** [review]
👍

**Review by Kay-Zee — APPROVED** [review]

**Pull request merged**